### PR TITLE
moving controllers to a struct to share between api/controller

### PR DESCRIPTION
**Pull request by luthermonson**
since the controllers need to be started to work in the api we need to block starting the controller now before we pass them into the api thusly i removed the enable-api/enable-controller flags for now

### CODE REVIEW

File: pkg/types/types.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
+package types
+
+import (
+	"github.com/rancher/terraform-controller/pkg/generated/controllers/terraformcontroller.cattle.io"
+	v1 "github.com/rancher/terraform-controller/pkg/generated/controllers/terraformcontroller.cattle.io/v1"
+	"github.com/rancher/wrangler/pkg/generated/controllers/batch"
+	v14 "github.com/rancher/wrangler/pkg/generated/controllers/batch/v1"
```
Comment (k0da) [inline pkg/types/types.go]: can you name it BatchV1?

File: pkg/types/types.go [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+	"github.com/rancher/wrangler/pkg/generated/controllers/core"
+	v13 "github.com/rancher/wrangler/pkg/generated/controllers/core/v1"
```
Comment (k0da) [inline pkg/types/types.go]: CoreV1?

File: pkg/types/types.go [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+	"github.com/rancher/wrangler/pkg/generated/controllers/rbac"
+	v12 "github.com/rancher/wrangler/pkg/generated/controllers/rbac/v1"
```
Comment (k0da) [inline pkg/types/types.go]: RbacV1?

File: pkg/types/types.go (hunk 1 above)
Comment (luthermonson) [reply pkg/types/types.go]: ya i can update

**Review by k0da — APPROVED** [review]
LGTM

**Pull request merged**